nixos/modules/services/hardware/udev.nix

```diff
@@ -8,6 +8,24 @@ let
 
   cfg = config.services.udev;
 
+  initrdUdevRules = pkgs.runCommand "initrd-udev-rules" {} ''
+    mkdir -p $out/etc/udev/rules.d
+    for f in 60-cdrom_id 60-persistent-storage 75-net-description 80-drivers 80-net-setup-link; do
+      ln -s ${config.boot.initrd.systemd.package}/lib/udev/rules.d/$f.rules $out/etc/udev/rules.d
+    done
+  '';
+
+
+  # networkd link files are used early by udev to set up interfaces early.
+  # This must be done in stage 1 to avoid race conditions between udev and
+  # network daemons.
+  # TODO move this into the initrd-network module when it exists
+  initrdLinkUnits = pkgs.runCommand "initrd-link-units" {} ''
```

Comment:
I don't understand this. We don't have to do this in the old initrd, why should we have to with this one?

Reply:
I just don't think we did link files in the old initrd. i.e. predictible interface names were not used.

Reply:
But can't we use the `initrd.systemd.units` logic for this?

Reply:
@arianvp That would place units in `/etc/systemd/system`, not in `/etc/systemd/network`.

Reply:
ah yes; but we have similar logic in the `networkd` module in stage-2. Perhaps we want to copy that to `stage-1` under `initrd.systemd.network.links` ? (https://search.nixos.org/options?channel=21.11&show=systemd.network.links&from=0&size=50&sort=relevance&type=packages&query=systemd.network.links)

Reply:
@arianvp see: https://github.com/NixOS/nixpkgs/pull/169116 :)

Reply:
So. we can keep this workaround for now; and then clean it up later once that PR is ready? Must just make sure to not forget.

Reply:
@ElvishJerricco we do this because the code was actually in the old stage 1 as well ;) It's not a new invention.
Old code: https://github.com/NixOS/nixpkgs/blob/b991bbd41e6d528fe0903d615a95301944b6e15b/nixos/modules/system/boot/stage-1.nix#L223

Reply:
Oh, heh cool. So yea, keep this, and revisit whenever we actually want to take that networkd patch seriously.